openedx/core/apidocs.py

```diff
@@ -0,0 +1,161 @@
+"""
+Open API support.
+"""
+
+import textwrap
+
+from drf_yasg import openapi
+from drf_yasg.generators import OpenAPISchemaGenerator
+from drf_yasg.utils import swagger_auto_schema as drf_swagger_auto_schema
+from drf_yasg.views import get_schema_view
+from rest_framework import permissions
+
+# -- Code that will eventually be in another openapi-helpers repo -------------
+
+
+class ApiSchemaGenerator(OpenAPISchemaGenerator):
+    """A schema generator for /api/*
+
+    Only includes endpoints in the /api/* url tree, and sets the path prefix
+    appropriately.
+    """
+
+    def get_endpoints(self, request):
+        endpoints = super(ApiSchemaGenerator, self).get_endpoints(request)
+        subpoints = {p: v for p, v in endpoints.items() if p.startswith("/api/")}
+        return subpoints
+
+    def determine_path_prefix(self, paths):
+        return "/api/"
+
+
+def dedent(text):
+    """
+    Dedent multi-line text nicely.
+
+    An initial empty line is ignored so that triple-quoted strings don't need
+    to start with a backslash.
+    """
+    if "\n" in text:
+        first, rest = text.split("\n", 1)
+        if not first.strip():
+            # First line is blank, discard it.
+            text = rest
+    return textwrap.dedent(text)
+
+
+def schema(
+    manual_parameters=None,
+    operation_summary=None,
+    operation_description=None
+):
```

Comment:
* How about `parameters` instead of `manual_parameters`? Just personal preference, up to you.
* Consider omitting `operation_summary` and `operation_description`, as the docstring style is much preferable.

Reply:
:+1: 
Omitting the `operation_*` parameters will simplify the code and resolve the pylint issues \o/